File: Soria_Christopher/native/ios/NativeAppearance/NativeAppearanceImpl.m
// ios/250App/NativeAppearance/NativeAppearanceImpl.m
#import "NativeAppearanceImpl.h"
#import <UIKit/UIKit.h>
#import <TargetConditionals.h>

@implementation NativeAppearanceImpl

+ (instancetype)shared {
  static NativeAppearanceImpl *s;
  static dispatch_once_t once;
  dispatch_once(&once, ^{ s = [NativeAppearanceImpl new]; });
  return s;
}

- (UIUserInterfaceStyle)styleFromString:(NSString *)str {
  NSString *s = str.lowercaseString;
  if ([s isEqualToString:@"dark"]) { return UIUserInterfaceStyleDark; }
  if ([s isEqualToString:@"light"]) { return UIUserInterfaceStyleLight; }
  return UIUserInterfaceStyleUnspecified;
}

- (void)setStyle:(NSString *)styleString {
  UIUserInterfaceStyle style = [self styleFromString:styleString];
  dispatch_async(dispatch_get_main_queue(), ^{
    // iOS 13+: connectedScenes → first window
    UIWindow *window = nil;
    NSSet *scenes = UIApplication.sharedApplication.connectedScenes;
    if (scenes.count > 0) {
      UIScene *scene = scenes.allObjects.firstObject;
      if ([scene isKindOfClass:UIWindowScene.class]) {
        window = ((UIWindowScene *)scene).windows.firstObject;
      }
    }
    if (window) {
      window.overrideUserInterfaceStyle = style;
    }
  });
}

- (NSString *)getCurrentStyle {
  NSSet *scenes = UIApplication.sharedApplication.connectedScenes;
  if (scenes.count > 0) {
    UIScene *scene = scenes.allObjects.firstObject;
    if ([scene isKindOfClass:UIWindowScene.class]) {
      UIWindow *window = ((UIWindowScene *)scene).windows.firstObject;
      if (window) {
        switch (window.overrideUserInterfaceStyle) {
          case UIUserInterfaceStyleDark: return @"dark";
          case UIUserInterfaceStyleLight: return @"light";
          default: return @"unspecified";
        }
      }
    }
  }
  return @"unspecified";
}

- (BOOL)isSimulator {
#if TARGET_OS_SIMULATOR
  return YES;
#else
  return NO;
#endif
}

@end

File: native/ios/NativeAppearance/NativeAppearanceImpl.swift
import UIKit

@objc(NativeAppearanceImpl)
public class NativeAppearanceImpl: NSObject {
  @objc public static let shared = NativeAppearanceImpl()

  private func style(from s: String) -> UIUserInterfaceStyle {
    switch s.lowercased() {
      case "dark": return .dark
      case "light": return .light
      default: return .unspecified
    }
  }

  @objc public func setStyle(_ s: String) {
    let ui = style(from: s)
    DispatchQueue.main.async {
      if let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
         let window = scene.windows.first {
        window.overrideUserInterfaceStyle = ui
      }
    }
  }

  @objc public func getCurrentStyle() -> String {
    if let scene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
       let window = scene.windows.first {
      switch window.overrideUserInterfaceStyle {
        case .dark: return "dark"
        case .light: return "light"
        default: break
      }
    }
    return "unspecified"
  }

  @objc public func isSimulator() -> Bool {
  #if targetEnvironment(simulator)
    return true
  #else
    return false
  #endif
  }
}